pkg/datasource.go

```diff
@@ -1,24 +1,49 @@
 package main
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
-	"golang.org/x/sync/errgroup"
-
-	"context"
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+	"golang.org/x/sync/errgroup"
+	"net/http"
 )
 
+func newResourceHandler() backend.CallResourceHandler {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/replace", func(w http.ResponseWriter, r *http.Request) {
+		replaceQuery(w, r)
+	})
+	mux.HandleFunc("/get-ast-property", func(w http.ResponseWriter, r *http.Request) {
+		getAstProperty(w, r)
+	})
+	mux.HandleFunc("/get-ast", func(w http.ResponseWriter, r *http.Request) {
+		getCompleteAst(w, r)
+	})
+	mux.HandleFunc("/apply-adhoc-filters", func(w http.ResponseWriter, r *http.Request) {
+		applyAdhocFilters(w, r)
+	})
+	mux.HandleFunc("/replace-time-filters", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
you can use function directly without wrappers
just `mux.HandleFunc("/replace-time-filters", replaceTimeFilters)`